apps/andi/lib/andi/migration/modified_date_migration.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Andi.Migration.ModifiedDateMigration do
+  @moduledoc """
+  For all existing dataset.business.modifiedDate, either parse to valid iso8601 or convert to empty string.
+  """
+
+  alias SmartCity.Dataset
+  import SmartCity.Event, only: [dataset_update: 0]
+
+  require Logger
+  require Andi
+  @instance Andi.instance_name()
+
+  def do_migration() do
+    Brook.get_all_values!(@instance, :dataset)
+    |> Enum.each(&migrate_dataset/1)
+  end
+
+  defp migrate_dataset(%Dataset{} = dataset) do
+    corrected_date = Andi.Migration.DateCoercer.coerce_date(dataset.business.modifiedDate)
+
+    if dataset.business.modifiedDate != corrected_date do
+      log_bad_date(dataset.id, dataset.business.modifiedDate, corrected_date)
+
+      updated_business =
+        dataset.business
+        |> Map.from_struct()
+        |> Map.put(:modifiedDate, corrected_date)
+
+      {:ok, updated_dataset} =
+        dataset
+        |> Map.from_struct()
+        |> Map.put(:technical, Map.from_struct(dataset.technical))
+        |> Map.put(:business, updated_business)
+        |> SmartCity.Dataset.new()
```

Comment:
This function is rather large, and it seems like this code could be pulled out into its own function or two.

Reply:
We've pulled it out. We'd like to perhaps create some helpers in smart_city so that we can more easily update a field on those structs without having to go through this (or implement the behaviors).